### ISSUE TITLE
Completely redo subreddit toolbox config.

### ISSUE THREAD

**Issue by creesch** [creesch/reddit-moderator-toolbox#355]
No milestone attached, but it is something that really needs to be done. 

There are many points where it doesn't function like you would expect it to based on how the rest of toolbox functions. This is the result of this being one of the more older and ancient modules, so nobody is at fault here but it is something we should get around at one point. 

Examples: 
- Tabs aren't really tabs, they where hacked in as a patch to make it somewhat look like the rest of toolbox.
- Edditing removal reasons and mod macros is simply odd. Editing those shouldn't involve overlaying windows but happen as part of the window on the tab. 
  - Having the "delete" and "save" buttons on the footer is bad UX considering that they don't save the entire page but the specific removal reason or macro on question. It makes more sense to make they specific buttons appear under the text areas. Or possibly to switch the list of reasons/macros with the editing areas. 
- The "save page to wiki" button randomly appears or stays on the window. 

Basically the first set of tabs for wikis is how it should always work. The second set of tabs is based on behavior that we should work to phase out. 

@dakta @agentlame @LowSociety @TheEnigmaBlade
